### PR TITLE
feat: Lazy-loading more information about members

### PR DESCRIPTION
**Pull request by RemiBardon**
TODO:

- [x] Nickname
- [ ] Presence (should I do it?)
  - Yes, in another PR
- [x] Avatar
- [x] OpenAPI
- [x] Smoke tests
- [x] Integration tests

(Merge #18 first)

### CODE REVIEW

File: src/v1/members/routes.rs [hunk 1]
```diff
@@ -37,6 +50,85 @@ pub(super) async fn get_members(
...
+            debug!("`{jid}` has no vCard.");
+            None
+        }
+        Err(err) => {
+            // Log error
+            error!("Could not get `{jid}`'s vCard: {err}");
+            // But dismiss it
+            None
+        }
+    };
```
Comment (RemiBardon) [inline src/v1/members/routes.rs]: We retrieve the user's vCard. If it fails (e.g. unknown user) or the user has no vCard, we do as if they had no vCard and return `null` fields.

File: src/v1/members/routes.rs [hunk 2]
```diff
@@ -37,6 +50,85 @@ pub(super) async fn get_members(
...
+            debug!("`{jid}` has no avatar.");
+            None
+        }
+        Err(err) => {
+            // Log error
+            error!("Could not get `{jid}`'s avatar: {err}");
+            // But dismiss it
+            None
+        }
+    };
```
Comment (RemiBardon) [inline src/v1/members/routes.rs]: Same if the user has no avatar, we return `null` (and print debug logs).

File: tests/integration/step-ci/members.yaml
Comment (RemiBardon) [inline tests/integration/step-ci/members.yaml:1]: The integration tests

Comment (RemiBardon) [inline tests/integration/step-ci/members.yaml:1]: The integration tests

File: static/api-docs/openapi.json
Comment (RemiBardon) [inline static/api-docs/openapi.json:1]: `source .env && cargo run` then open <http://localhost:8000/api-docs/redoc#tag/Members/operation/enrich_members> to see the API docs

File: src/v1/members/routes.rs [hunk 3]
```diff
@@ -37,6 +50,85 @@ pub(super) async fn get_members(
...
+            debug!("`{jid}` has no avatar.");
+            None
+        }
+        Err(err) => {
+            // Log error
+            warn!("Could not get `{jid}`'s avatar: {err}");
+            // But dismiss it
+            None
+        }
+    };
```
Comment (RemiBardon) [inline src/v1/members/routes.rs:85]: Same if the user has no avatar, we return `null` (and print debug logs).

File: src/v1/members/routes.rs [hunk 4]
```diff
@@ -37,6 +50,85 @@ pub(super) async fn get_members(
...
+            debug!("`{jid}` has no vCard.");
+            None
+        }
+        Err(err) => {
+            // Log error
+            warn!("Could not get `{jid}`'s vCard: {err}");
+            // But dismiss it
+            None
+        }
+    };
```
Comment (RemiBardon) [inline src/v1/members/routes.rs:68]: We retrieve the user's vCard. If it fails (e.g. unknown user) or the user has no vCard, we do as if they had no vCard and return `null` fields.

**Comment by RemiBardon** [comment]
@valeriansaliou I don't understand why GitHub marked so many review comments as outdated even though the files didn't change. After we review in the file explorer, we should come back here and resolve the remaining conversations.

**Review by valeriansaliou — APPROVED** [review]

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -92,6 +92,10 @@ make integration-test
 
 If a test fails, Step CI will automatically print some additional information to help you debug the issue. We also print container logs so you can see internal errors.
 
+> [!TIP]
```
[the remark this answers is not in the document]
Comment (RemiBardon) [reply CONTRIBUTING.md:95]: > Enable by defaults

**Pull request merged**